crates/xflags/src/lib.rs

```diff
@@ -162,6 +162,14 @@
 //! }
 //! ```
 //!
+//! You can create aliases if desired. In this case, `run` can be called as `run`, `r` and `exec`:
+//! 
+//! ```rust
+//! xflags::xflags! {
+//!     cmd run alias r exec {}
```

Comment:
yeah that seems perfectly reasonable. In that case, maybe rather than having a `.name` and a `.aliases` attribute, we could use a single vec and access `.names[0]` to get the main name?

Reply:
That is not public API, so that it doesn't matter really, but I do prefer`name/aliases` there:

```rust
struct NonEmpty {
  first: X
  rest: Vec<X>
}
```

is a nice idiom to enforce that the list is non-empty, and we do want to do it here (`cmd {}` without a name should be an error, would be good to add a test for that as well). 

Besides, the first alias is biased is that it, unlike all the others, gives the name to the rust struct, so that's another reason why it should be a separate field.